pkgs/applications/networking/instant-messengers/discord/default.nix

```diff
@@ -1,35 +1,89 @@
-{ branch ? "stable", pkgs }:
+{ branch ? "stable", pkgs, lib, stdenv }:
 let
   inherit (pkgs) callPackage fetchurl;
-in {
-  stable = callPackage ./base.nix rec {
-    pname = "discord";
-    binaryName = "Discord";
-    desktopName = "Discord";
-    version = "0.0.16";
-    src = fetchurl {
-      url = "https://dl.discordapp.net/apps/linux/${version}/discord-${version}.tar.gz";
-      sha256 = "UTVKjs/i7C/m8141bXBsakQRFd/c//EmqqhKhkr1OOk=";
-    };
+  versions = if stdenv.isLinux then {
+    stable = "0.0.16";
+    ptb = "0.0.27";
+    canary = "0.0.132";
+  } else {
+    stable = "0.0.264";
+    ptb = "0.0.58";
+    canary = "0.0.280";
   };
-  ptb = callPackage ./base.nix rec {
-    pname = "discord-ptb";
-    binaryName = "DiscordPTB";
-    desktopName = "Discord PTB";
-    version = "0.0.27";
-    src = fetchurl {
-      url = "https://dl-ptb.discordapp.net/apps/linux/${version}/discord-ptb-${version}.tar.gz";
-      sha256 = "0yphs65wpyr0ap6y24b0nbhq7sm02dg5c1yiym1fxjbynm1mdvqb";
+  version = versions.${branch};
+  srcs = {
+    x86_64-linux = {
+      stable = fetchurl {
+        url =
+          "https://dl.discordapp.net/apps/linux/${version}/discord-${version}.tar.gz";
+        sha256 = "UTVKjs/i7C/m8141bXBsakQRFd/c//EmqqhKhkr1OOk=";
+      };
+      ptb = fetchurl {
+        url =
+          "https://dl-ptb.discordapp.net/apps/linux/${version}/discord-ptb-${version}.tar.gz";
+        sha256 = "0yphs65wpyr0ap6y24b0nbhq7sm02dg5c1yiym1fxjbynm1mdvqb";
+      };
+      canary = fetchurl {
+        url =
+          "https://dl-canary.discordapp.net/apps/linux/${version}/discord-canary-${version}.tar.gz";
+        sha256 = "1jjbd9qllgcdpnfxg5alxpwl050vzg13rh17n638wha0vv4mjhyv";
+      };
+    };
+    x86_64-darwin = {
+      stable = fetchurl {
+        url = "https://dl.discordapp.net/apps/osx/${version}/Discord.dmg";
+        sha256 = lib.fakeSha256;
```

Comment:
Could someone on x86_64-darwin send the hash for this

Reply:
you can compute this locally:
```
$ nix-prefetch-url "https://dl-canary.discordapp.net/apps/osx/0.0.264/DiscordCanary.dmg"
path is '/nix/store/nw6pavknfjy3l4z4kl3rwn7z5qk73him-DiscordCanary.dmg'
10gagrrnf1vbj8fj7b0gpqxpclgphy005aggsd76mwny9zij06lm
```

Reply:
You can also do something like
```
nix-instantiate --eval -E 'with import ./. { system = "x86_64-darwin"; }; lib.head discord.src.urls'
```
to figure our the urls

Reply:
Thanks!